packages/amplify_auth_cognito/lib/method_channel_auth_cognito.dart

```diff
@@ -419,11 +420,18 @@ class AmplifyAuthCognitoMethodChannel extends AmplifyAuthCognito {
 
   UpdateUserAttributeResult _formatUpdateUserAttributeResponse(
       Map<String, dynamic> res) {
+    var codeDeliveryDetails = res["nextStep"]["codeDeliveryDetails"];
     return UpdateUserAttributeResult(
         isUpdated: res["isUpdated"],
         nextStep: AuthNextUpdateAttributeStep(
             updateAttributeStep: res["nextStep"]["updateAttributeStep"],
-            codeDeliveryDetails: res["nextStep"]["codeDeliveryDetails"],
+            codeDeliveryDetails: codeDeliveryDetails != null
+                ? AuthCodeDeliveryDetails(
+                    attributeName: codeDeliveryDetails["attributeName"] ?? null,
+                    deliveryMedium:
+                        codeDeliveryDetails["deliveryMedium"] ?? null,
+                    destination: codeDeliveryDetails["destination"])
```

Comment:
This is throwing an error, `type 'Null' is not a subtype of type 'String'`. You are checking that codeDeliveryDetails != null, but `codeDeliveryDetails["destination"]` can still be null. I am guessing this is the main reason you changed the kotlin/swift side and added that util (to consistently return nulls, instead of empty maps)?

This can be fixed by updating the swift/kotlin side for FlutterUpdateUserAttributeResult in the same way that the other results (signUp, signIn, resetPassword, etc.) were updated so that it returns null instead of an empty map. This should be straightforward now, but a heads up that it will cause this PR to conflict with https://github.com/aws-amplify/amplify-flutter/pull/601. My first thought was that I could pull some of these changes into https://github.com/aws-amplify/amplify-flutter/pull/601 so that we don't have to worry about conflicts later but I don't think I can really do that since even just the changes on the swift/kt side could be considered breaking (returning null instead of an empty map). If we go this route, I think we will have to just reconcile some conflicts/inconsistencies when you next pull master into this branch.

Another option would be to add checks for empty maps (or null `codeDeliveryDetails["destination"]`) on the dart side (which maybe should be added anyway?) and then I think the changes on the swift/kt side would not be needed. Returning nulls instead of empty maps and consolidating the logic makes sense but it seems a little out of scope for dart null safety upgrade and if I am understanding the purpose of the changes, it seems easy enough to resolve on the dart side.

It is worth noting that with either approach there is more work that should eventually be done to consolidate logic and return more logical values across auth.

Reply:
Thanks.  @haverchuck I noticed we made destination field of AuthCodeDeliveryDetails class required, can we make it nullable like all of the other fields in that class?

Reply:
@Jordan-Nelson Which method is throwing this error?

Reply:
`updateUserAttribute()`. Only on iOS when you update an attribute that doesn't have a confirmation code (anything other than email/phone)

Reply:
Heh - sorry I realized just now I could have seen this for myself.  This makes sense. I'm taking a look.